src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -524,3 +534,58 @@ def __publish_new_namespaces(self, namespaces: List[NamespaceInfo]):
 
     def get_global_config(self) -> dict:
         return self.registry.get_global_config()
+
+    def __update_node(self, new_node: Node, operation: K8sOperationType):
+        # cant get the extra pods discovery info.
+        if operation == K8sOperationType.CREATE:
+            return
+
+        with self.services_publish_lock:
+            name = new_node.metadata.name
+            cache = self.__nodes_cache.get(name, None)
+            if cache is None:
+                return
+
+            if cache.resource_version >= int(new_node.metadata.resourceVersion or 0):
+                return
+
+            new_info = self.__from_api_server_node(new_node, [])
+            new_info.memory_allocated = cache.memory_allocated
+            new_info.cpu_allocated = cache.cpu_allocated
+            new_info.pods_count = cache.pods_count
+            new_info.pods = cache.pods
+
+            if operation == K8sOperationType.UPDATE:
+                self.__nodes_cache[name] = new_info
+            elif operation == K8sOperationType.DELETE:
+                self.__services_cache.pop(name, None)
+                new_info.deleted = True
+
+            self.dal.publish_nodes([new_info])
+            self.__discovery_metrics.on_nodes_updated(1)
+
+    def __update_job(self, new_job: Job, operation: K8sOperationType):
+
+        new_info = JobInfo.from_api_server(new_job, [])
+        job_key = new_info.get_service_key()
+        with self.services_publish_lock:
+            if operation == K8sOperationType.UPDATE:
+                old_info = self.__jobs_cache.get(job_key, None)
+                if old_info is None:  # Update may occur after delete.
+                    return
+
+                new_info.job_data = old_info.job_data
```

Comment:
can the new_info have different job data that isn't in old info, like pods? (other fields in data might not be mutable )